frame/elections-phragmen/src/election.rs

```diff
@@ -0,0 +1,150 @@
+
+		match T::votes_with_stake().into_iter().try_for_each(
+			|(voter, crate::Voter { stake, votes, .. })| {
+				if let Some(max_voters) = maybe_max_len {
+					if voters_and_stakes.len() > max_voters {
+						return Err(());
+					}
+				}
+				let bounded_votes: BoundedVec<_, T::MaxVotesPerVoter> =
+					BoundedVec::try_from(votes).map_err(|_| ())?; // TODO(gpestana): ref to proper err
+
+				voters_and_stakes.push((voter, stake, bounded_votes));
+				Ok(())
+			},
+		) {
+			Ok(_) => (),
+			Err(_) => return Err(""), // TODO(gpestana): ref proper err
+		}
+
+		Ok(voters_and_stakes)
+	}
+
+	fn desired_targets() -> frame_election_provider_support::data_provider::Result<u32> {
+		// TODO(gpestana): fill in
+		Ok(10)
+	}
+
+	fn next_election_prediction(now: Self::BlockNumber) -> Self::BlockNumber {
+		// TODO(gpestana): fill in
+		<frame_system::Pallet<T::System>>::block_number()
+	}
+}
+
+pub trait ElectionConfig {
+	type System: frame_system::Config;
+	type DataProvider: ElectionDataProvider<
+		AccountId = AccountId<Self::System>,
+		BlockNumber = BlockNumber<Self::System>,
+	>;
+	type Solver: NposSolver<AccountId = AccountId<Self::System>, Error = sp_npos_elections::Error>;
+	type WeightInfo: WeightInfo;
+}
+
+pub struct BoundedExecution<T: ElectionConfig>(PhantomData<T>);
+
+fn elect_with_bounds<T: ElectionConfig>(
+	maybe_max_voters: Option<usize>,
+	maybe_max_targets: Option<usize>,
+) -> Result<Supports<AccountId<T::System>>, Error> {
+	// TODO(gpestana): finsh impl
+
+	// TODO(gpestana): handle unwraps()
+	let num_to_elect = T::DataProvider::desired_targets().unwrap();
+	let candidate_ids = T::DataProvider::electable_targets(maybe_max_targets).unwrap();
+	let voters_and_votes = T::DataProvider::electing_voters(maybe_max_voters).unwrap();
+
+	T::Solver::solve(num_to_elect as usize, candidate_ids, voters_and_votes)
+		.map(|election_result| {});
+
+	Ok(vec![])
+}
+
+impl<T: ElectionConfig> ElectionProviderBase for BoundedExecution<T> {
+	type AccountId = AccountId<T::System>;
+	type BlockNumber = BlockNumber<T::System>;
+	type Error = Error;
+	type DataProvider = T::DataProvider;
+
+	fn ongoing() -> bool {
+		return false;
+	}
+}
+
+impl<T: ElectionConfig> ElectionProvider for BoundedExecution<T> {
+	fn elect() -> Result<frame_election_provider_support::Supports<Self::AccountId>, Self::Error> {
+		// This should not be called if not in `std` mode (and therefore neither in genesis nor in
+		// testing)
+		if cfg!(not(feature = "std")) {
+			frame_support::log::error!(
+				"Please use `InstantElectionProvider` instead to provide bounds on election if not in \
+				genesis or testing mode"
+			);
+		}
+
+		elect_with_bounds::<T>(None, None)
+	}
+}
+
+impl<T: ElectionConfig> InstantElectionProvider for BoundedExecution<T> {
+	fn elect_with_bounds(
+		max_voters: usize,
+		max_targets: usize,
+	) -> Result<Supports<Self::AccountId>, Self::Error> {
+		elect_with_bounds::<T>(Some(max_voters), Some(max_targets))
+	}
+}
```

Comment:
Having a look now, i don't see a need for anything declared in this file, to the best of my knowledge.